cpp/src/structs/utilities.cpp

```diff
@@ -61,6 +63,27 @@ std::vector<std::vector<column_view>> extract_ordered_struct_children(
   return result;
 }
 
+namespace {
+/**
+ * @brief Check whether the specified column is of type `STRUCT`.
+ *
+ */
+bool is_struct(cudf::column_view const& col) { return col.type().id() == type_id::STRUCT; }
+
+/**
+ * @brief Check whether the specified column is of type LIST,
+ *        or any LISTs in its descendent columns.
+ *
+ */
+bool is_or_has_lists(cudf::column_view const& col)
+{
+  auto is_list = [](cudf::column_view const& col) { return col.type().id() == type_id::LIST; };
+
+  return is_list(col) ||
+         (is_struct(col) && std::any_of(col.child_begin(), col.child_end(), is_or_has_lists));
+}
```

Comment:
```suggestion
bool has_list_nested(cudf::column_view const& col)
{
  auto is_list = [](cudf::column_view const& col) { return col.type().id() == type_id::LIST; };

  return is_list(col) ||
         (is_struct(col) && std::any_of(col.child_begin(), col.child_end(), is_or_has_lists));
}
```

Reply:
I also wonder why do we need to check `is_struct(col)` before checking if the column children has lists? Can we just:
```
return is_list(col) || std::any_of(col.child_begin(), col.child_end(), has_list_nested);
```

Reply:
`has_list_nested()` would not be an accurate name, because this function also checks whether the current column is a list.

You might have a point about the `is_struct` check, although it does mean that we will iterate over the children of `string` columns. I would prefer to keep this function as it stands.

Reply:
No, what you said should have the name `has_child_list` instead. `has_list_nested` or `has_nested_list` should mean we have list at any level. 
We already have this `has_nested` function:
```
inline bool has_nested_nulls(table_view const& input)
{
  return std::any_of(input.begin(), input.end(), [](auto const& col) {
    return col.has_nulls() ||
           std::any_of(col.child_begin(), col.child_end(), [](auto const& child_col) {
             return has_nested_nulls(table_view{{child_col}});
           });
  });
}
```

Reply:
In addition, iterate over children of strings column will return false --- there is not any list in it. So we are safe to just call `has_nested_list` on the children columns of any column.

Reply:
Good idea about adding `nested` to the function name. We're checking grandchildren, great-grandchildren, etc.

> there is not any list in it. 

Not questioning that. I wanted to avoid iterating on children of columns that could not have list children. That said, I'm happy to adopt your suggestion:
1. `is_struct()` isn't checked now.
2. The function now has `nested` in the name.

I'm retaining `is_or_has`, because those are different conditions, and we're checking both. `has_nested_lists` would imply that we're checking the children, not the column itself. Agreeable?